### PR TITLE
Ability to upload extra documentation to Exchange

### PR DESCRIPTION
**Pull request by benrudolph**
@esoergel 

https://trello.com/c/GmllK6Sc/9-exchange-ability-to-publish-documentation-alongside-an-app
https://trello.com/c/BsLAyK0F/314-upload-documentation-to-the-exchange

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/appstore/urls.py, line 24 - E501 line too long (129 > 115 characters)
* corehq/apps/domain/tests/test_views.py, line 28 - W291 trailing whitespace
* corehq/apps/domain/tests/test_views.py, line 29 - W291 trailing whitespace
* corehq/apps/domain/tests/test_views.py, line 89 - E303 too many blank lines (2)
* corehq/apps/domain/views.py, line 1749 - E501 line too long (145 > 115 characters)

File: corehq/apps/domain/views.py [hunk 1]
```diff
@@ -1618,8 +1618,10 @@ def fixture_forms(self):
     @memoized
     def snapshot_settings_form(self):
         if self.request.method == 'POST':
-            form = SnapshotSettingsForm(self.request.POST, self.request.FILES)
-            form.dom = self.domain_object
```
Comment (esoergel) [inline corehq/apps/domain/views.py]: good call, I much prefer passing the domain in when you init the form.

Comment (benrudolph) [inline corehq/apps/domain/views.py]: haha i think that was your call when we were pairing on this :)

Comment (esoergel) [inline corehq/apps/domain/views.py]: oh really? whoops, I'd forgotten about this

File: corehq/apps/domain/forms.py [hunk 2]
```diff
@@ -169,9 +169,13 @@ class SnapshotSettingsForm(SnapshotSettingsMixin):
         help_text=ugettext_noop("An optional image to show other users your logo or what your app looks like"))
     video = CharField(label=ugettext_noop("Youtube Video"), required=False,
         help_text=ugettext_noop("An optional youtube clip to tell users about your app. Please copy and paste a URL to a youtube video"))
+    documentation_file = forms.FileField(label=ugettext_noop("Documentation File"), required=False,
+        help_text=ugettext_noop("An optional file to tell users more about your app."))
     cda_confirmed = BooleanField(required=False, label=ugettext_noop("Content Distribution Agreement"))
 
     def __init__(self, *args, **kw):
+        self.dom = kw.pop("domain", None)
```
Comment (millerdev) [inline corehq/apps/domain/forms.py]: Is there a reason why you shortened "domain" to "dom"? When I see "dom" I think Document Object Model.

Comment (benrudolph) [inline corehq/apps/domain/forms.py]: Agreed dom is misleading, but it was already being used as dom: https://github.com/dimagi/commcare-hq/pull/5689/files#diff-21bd1c0e52720275d9cf19e3dd556434L1622. Presumably something depends on it being called dom. I think it should be changed but probably in a separate PR
